### PR TITLE
WR309191 add redis check

### PR DESCRIPTION
**Pull request by tomdickman**
This improvement adds an explicit Redis check which, if Redis is the Session handler for Moodle, creates a connection (outside of Moodle APIs and libraries, in order to remove this as a variability for failure) and sends a timestamped key value pair to the Redis store, before retrieving and checking the value, updating the status if passing or calling a fail function if not.

### CODE REVIEW

File: lib.php [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+ * @copyright  2019 Tom Dickman <tomdickman@catalyst-au.net>
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+/**
+ * Return an error that ELB will pick up
+ *
+ * @param string $reason
+ */
+function failed($reason) {
```
Comment (brendanheywood) [inline lib.php]: this one is in nagios.lib now

File: lib.php [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    exit;
+}
+
+/**
+ * Checks if the command line maintenance mode has been enabled. Skip the config bootstrapping.
+ *
+ * @param string $configfile The relative path for config.php
+ * @return bool True if climaintenance.html is found.
+ */
+function check_climaintenance($configfile) {
```
Comment (brendanheywood) [inline lib.php]: lib.php should be minimal and for callbacks, I think this should be called locallib.php instead

File: lib.php (hunk 1 above)
Comment (tomdickman) [reply lib.php]: Removed and replaced with calls to nagios functions throughout index.php.

File: lib.php (hunk 2 above)
Comment (tomdickman) [reply lib.php]: moved to locallib.php

**Pull request closed without merging**